spec/feature/login_spec.rb

```diff
@@ -190,32 +190,76 @@
       )
     end
 
-    before do
-      expect(user.ephemeral_states.last).to be_nil
+    context "when the feature_flag_enforce_levels_of_authentication is not 'enabled'" do
+      before do
+        expect(user.ephemeral_states.last).to be_nil
+        log_in
+      end
 
-      visit authorization_endpoint_url(client: application, scope: "openid")
-      user_is_returned_to_login_screen
+      it "records that the user has logged in with level-of-authentication 1" do
+        enter_mfa
 
-      fill_in "email", with: user.email
-      fill_in "password", with: user.password
-      click_on I18n.t("devise.sessions.new.fields.submit.label")
-    end
+        expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level1")
+      end
 
-    it "records that the user has logged in with level-of-authentication 1" do
-      enter_mfa
+      context "when the user doesn't have MFA set up" do
+        let(:user) { FactoryBot.create(:user, :without_mfa) }
 
-      expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level1")
+        it "records that the user has logged in with level-of-authentication 0" do
+          expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level0")
+        end
+      end
     end
 
-    context "when the user doesn't have MFA set up" do
-      let(:user) { FactoryBot.create(:user, :without_mfa) }
+    context "when the feature_flag_enforce_levels_of_authentication is 'enabled'" do
+      context "for a level0 authorised user" do
+        before do
+          allow(Rails.configuration).to receive(:feature_flag_enforce_levels_of_authentication).and_return(true)
+          application.update!(scopes: "openid level0")
+          log_in(%w[level0])
+        end
+
+        let(:user) { FactoryBot.create(:user, :without_mfa) }
```

Comment:
A user with MFA configured should also be able to log in at `level0`, so remote this `let` (and the one on line 240 is redundant, so remove it too)

Reply:
Oh I see, this is to test that the error is thrown - never mind.